packages/broker/src/Plugin.ts

```diff
@@ -17,6 +17,7 @@ export interface PluginOptions {
     apiAuthenticator: ApiAuthenticator
     metricsContext: MetricsContext
     brokerConfig: Config
+    storageNodeRegistry: StorageNodeRegistryItem[]
```

Comment:
Would it make sense for the `broker.ts`  main script to create an instance of `StorageNodeRegistry` and then pass the shared instance to the plugins instead of having each plugin create their own instance. It should work nicely since it is a rather immutable class. Also ping @teogeb .